synchronizer/synchronizer.go

```diff
@@ -682,3 +730,113 @@ func (s *ClientSynchronizer) processVerifiedBatch(verifiedBatch etherman.Verifie
 		log.Fatalf("error storing the verifiedBatch in processVerifiedBatch. BlockNumber: %d, error: %s", verifiedBatch.BlockNumber, err.Error())
 	}
 }
+
+func (s *ClientSynchronizer) processTrustedBatch(trustedBatch *pb.GetBatchResponse, dbTx pgx.Tx) error {
+	log.Debugf("processing trusted batch: %v", trustedBatch.BatchNumber)
+	processCtx := state.ProcessingContext{
+		BatchNumber:    trustedBatch.BatchNumber,
+		Coinbase:       common.HexToAddress(trustedBatch.Sequencer),
+		Timestamp:      time.Unix(int64(trustedBatch.Timestamp), 0),
+		GlobalExitRoot: common.HexToHash(trustedBatch.GlobalExitRoot),
+	}
+	txs := []types.Transaction{}
+	for _, transaction := range trustedBatch.Transactions {
+		tx, err := state.DecodeTx(transaction.Encoded)
+		if err != nil {
+			return err
+		}
+		txs = append(txs, *tx)
+	}
+	trustedBatchL2Data, err := state.EncodeTransactions(txs)
+	if err != nil {
+		return err
+	}
+
+	batch, err := s.state.GetBatchByNumber(s.ctx, trustedBatch.BatchNumber, nil)
+	if err != nil && err != state.ErrStateNotSynchronized {
+		log.Warnf("failed to get batch %v from local trusted state. Error: %v", trustedBatch.BatchNumber, err)
+		return err
+	}
+
+	// check if batch needs to be synchronized
+	if batch != nil {
+		matchNumber := batch.BatchNumber == trustedBatch.BatchNumber
+		matchGER := batch.GlobalExitRoot.String() == trustedBatch.GlobalExitRoot
+		matchLER := batch.LocalExitRoot.String() == trustedBatch.LocalExitRoot
+		matchSR := batch.StateRoot.String() == trustedBatch.StateRoot
+		matchCoinbase := batch.Coinbase.String() == trustedBatch.Sequencer
+		matchTimestamp := uint64(batch.Timestamp.Unix()) == trustedBatch.Timestamp
+		matchL2Data := hex.EncodeToString(batch.BatchL2Data) == hex.EncodeToString(trustedBatchL2Data)
+
+		if matchNumber && matchGER && matchLER && matchSR &&
+			matchCoinbase && matchTimestamp && matchL2Data {
+			log.Debugf("batch %v already synchronized", trustedBatch.BatchNumber)
+			return nil
+		}
+		log.Infof("batch %v needs to be updated", trustedBatch.BatchNumber)
+	} else {
+		log.Infof("batch %v needs to be synchronized", trustedBatch.BatchNumber)
+	}
+
+	log.Debugf("resetting trusted state from batch %v", trustedBatch.BatchNumber)
+	if err := s.state.ResetTrustedState(s.ctx, trustedBatch.BatchNumber-1, dbTx); err != nil {
+		log.Errorf("failed to reset trusted state", trustedBatch.BatchNumber)
+		return err
+	}
+
+	log.Debugf("opening batch %v", trustedBatch.BatchNumber)
+	if err := s.state.OpenBatch(s.ctx, processCtx, dbTx); err != nil {
+		log.Errorf("error opening batch %d", trustedBatch.BatchNumber)
+		return err
+	}
```

Comment:
`ProcessSequencerBatch` which you are calling just bellow already takes care of opening the batch

Reply:
where?